config/config.example.yml

```diff
@@ -42,26 +44,67 @@ services:
     priority: 1
     workspace_dir: ${WORKSPACE_DIR}
 
-# [Required] This section defines how to synchronize permissions between services when they shared resources
+# [Required] This section defines how to synchronize permissions between services when they share resources
 sync_permissions:
   # Friendly name to identify a sync point (The value is not used by Cowbird so this can be any relevant keyword)
   user_workspace:
     # [Required] This section defines a list of services (referring to the service name of the 'services' section)
     # sharing resources
-    services:
-      Geoserver: /api/workspaces/private/                # <Service name>:<Resource root matching with other services>
-      Thredds: /catalog/birdhouse/workspaces/private/    # Every service sharing the same resources can be added in this
-                                                         # list
+    services: # Contains the different resources that can be synchronized, ordered by service
+      Geoserver: # Service name, which should also be found in the services list at the top of this file
+        # Resource key (ex.: Geoserver_workspace): Custom name to represent a resource path.
+        # It should correspond to the names used in the `permission_mappings` section below.
+        # It contains the list of each segment of its resource path, with their corresponding name and type.
+        # The name of a segment can either contain a string name, a `SINGLE_TOKEN` `*` (to represent
+        # a single repetition of any name with the corresponding type) or a `MULTI_TOKEN` `**` (to
+        # represent any number (0 or more) of names that fit with the corresponding type). Note that tokens can
+        # only be found at the end of the path, so no named string should follow a token.
+        # Also, a MULTI_TOKEN can only be used one time in each list of path segments.
```

Comment:
So, there are some validations about the use of tokens. Tokens can only be found at the end of a path, and `**` can only be used one time, else it becomes too ambiguous.
So the example `/path/**/other/*` would result in an error since the named segment `other` is found after a token, but we could have `/path/**/*/*` for example.

Reply:
OK. I didn't even think about re-using `*`. If this is permitted, maybe add it in the comment as a possible workaround if this kind of specific combination is needed.